libraries/rush-lib/src/cli/actions/BaseInstallAction.ts

```diff
@@ -142,23 +148,38 @@ export abstract class BaseInstallAction extends BaseRushAction {
     }
 
     // If we are doing a filtered install and subspaces is enabled, we need to find the affected subspaces and install for all of them.
-    let selectedSubspaces: ReadonlySet<Subspace> | undefined;
-    const filteredProjectsForSubspace: Map<Subspace, RushConfigurationProject[]> = new Map();
+    let selectedSubspaces: Set<Subspace> | undefined;
+    const selectedProjectsMetadataBySubspace: Map<Subspace, ISubspaceSelectedProjectsMetadata> = new Map();
     if (this.rushConfiguration.subspacesFeatureEnabled) {
-      if (installManagerOptions.filteredProjects.length) {
-        // Go through each project, add it to it's subspace's pnpm filter arguments
-        for (const project of installManagerOptions.filteredProjects) {
-          let subspaceFilteredProjects: RushConfigurationProject[] | undefined =
-            filteredProjectsForSubspace.get(project.subspace);
-          if (!subspaceFilteredProjects) {
-            subspaceFilteredProjects = [];
-            filteredProjectsForSubspace.set(project.subspace, subspaceFilteredProjects);
+      selectedSubspaces = new Set();
+      const { selectedProjects } = installManagerOptions;
+      if (selectedProjects.size !== this.rushConfiguration.projects.length) {
+        // This is a filtered install. Go through each project, add its subspace's pnpm filter arguments
+        for (const project of selectedProjects) {
+          const { subspace: projectSubspace } = project;
```

Comment:
Should be using `getSubspacesForProjects` defined above to avoid duplicate processing of the metadata.